Add script to prepare dataset from csv (#462)
Lightning-AI/lit-gpt: tutorials/prepare_dataset.md

```
(Note that epending on the task, the `"input"` text can be an empty string, as shown above.)

Custom datasets can be prepared by either creating a new `scripts/prepare_dataset.py` script or reading the dataset
from a CSV file.

&nbsp;

### Preparing Custom Datasets From a CSV File

If you have a CSV file containing the following columns

- `instruction`: Column which will describe the task.
- `input`: A string holding a special input value for the instruction. This applies to some samples, and in others, this is empty (empty string).
- `output`: The expected response string.

If any of the columns is missing, then the script will fail to create the dataset. 

Before starting to finetune, you need to read, tokenize, and write the data converted from the CSV in a binary format. The simplest way to prepare the dataset is by simply running:

```bash

> python scripts/prepare_csv.py path/to/the/file.csv

Reviewer: ```suggestion
python scripts/prepare_csv.py --csv_path path/to/the/file.csv
```

```
You can also customize the dataset generation by using these additional parameters

- `destination_path`: The folder where the binary data will be saved. By default, it is saved inside `data/csv`

- `checkpoint_dir`: The model checkpoint dir. It will use the model's tokenizer to load and convert the string to input ids. Defaults to `"checkpoints/stabilityai/stablelm-base-alpha-3b"`

- `test_split_fraction`: The fraction of the data to split. Defaults to `0.1`

- `seed`: The seed value to reproduce the same random splits for train and test data.

- `mask_inputs`: Whether we require any masking or not.

- `ignore_index`: Mask out all the tokens after this index when preparing the dataset.

To use the the settings described above, you can add the respective command line arguments when calling `prepare_csv.py` as shown in the example below:

```bash

> python scripts/prepare_csv.py test_data.csv \

Reviewer: ```suggestion
python scripts/prepare_csv.py --csv_path test_data.csv \
```

```
(Note that epending on the task, the `"input"` text can be an empty string, as shown above.)

Custom datasets can be prepared by either creating a new `scripts/prepare_dataset.py` script or reading the dataset
from a CSV file.

&nbsp;

### Preparing Custom Datasets From a CSV File

If you have a CSV file containing the following columns

- `instruction`: Column which will describe the task
- `input`: A string holding a special input value for the instruction. This applies to some samples, and in others, this is empty (empty string).

> - `output`: The expected response string

Reviewer 2: I would end all entries here with a period "." for consistency.

If any of the columns is missing, then the script will fail to create the dataset. 

Before starting to finetune, you need to read, tokenize, and write the data converted from the CSV in a binary format. The simplest way to prepare the dataset is by simply running:

```bash

> python scripts/prepare_csv.py /path/to/the/.csv file

Reviewer: ```suggestion
python scripts/prepare_csv.py path/to/the/file.csv
```

```

> An example:
> 
> ```bash
> python scripts/prepare_csv.py /home/user/Downloads/data.csv
> ```

Reviewer: Repeating this is probably unnecessary, it should be clear from the bash snippet above

```suggestion
```

You can also customize the dataset generation by using these additional parameters

- `destination_path`: The folder where the binary data will be saved. By default, it is saved inside `data/csv`

- `checkpoint_dir`: The model checkpoint dir. It will use the model's tokenizer to load and convert the string to input ids. Defaults to `"checkpoints/stabilityai/stablelm-base-alpha-3b"`

- `test_split_fraction`: The fraction of the data to split. Defaults to `0.1`

- `seed`: The seed value to reproduce the same random splits for train and test data.

- `mask_inputs`: Whether we require any masking or not.

> - `ignore_index`: Explicitely which index to ignore when preparing the dataset. 

Reviewer: ```suggestion
- `ignore_index`: Mask out all the tokens after this index when preparing the dataset.
```
